rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -177,105 +176,94 @@ else()
     ${_generated_msg_py_files}
     ${_generated_msg_c_files}
     ${_generated_srv_files}
+    ${_generated_extension_files}
+    ${_extension_dependencies}
+    ${rosidl_generate_interfaces_TARGET}__rosidl_generator_c
   )
 endif()
 
 macro(set_properties _build_type)
-  set_target_properties(${_msg_name}${_pyext_suffix} PROPERTIES
+  set_target_properties(${_target_name} PROPERTIES
     COMPILE_FLAGS "${_extension_compile_flags}"
     PREFIX ""
-    LIBRARY_OUTPUT_DIRECTORY${_build_type} "${_output_path}/${_parent_folder}"
-    RUNTIME_OUTPUT_DIRECTORY${_build_type} "${_output_path}/${_parent_folder}"
-    OUTPUT_NAME "${_base_msg_name}__${_typesupport_impl}${PythonExtra_EXTENSION_SUFFIX}"
+    LIBRARY_OUTPUT_DIRECTORY${_build_type} ${_output_path}
+    RUNTIME_OUTPUT_DIRECTORY${_build_type} ${_output_path}
+    OUTPUT_NAME "${PROJECT_NAME}_s__${_typesupport_impl}${PythonExtra_EXTENSION_SUFFIX}"
```

Comment:
Unfortunately, the filename of the Python extension must match the name of the module in https://github.com/ros2/rosidl/blob/add_nested_msg_support/rosidl_generator_py/resource/_msg_support.entry_point.c.template#L32, so you can't a different value for `${PythonExtra_EXTENSION_SUFFIX}` other than the empty string on Windows.